reporter/src/main/kotlin/reporters/freemarker/FreemarkerTemplateProcessor.kt

```diff
@@ -393,6 +394,15 @@ class FreemarkerTemplateProcessor(
          */
         fun advisorResultsWithDefects(): Map<Identifier, List<AdvisorResult>> =
             input.filteredAdvisorResults(AdvisorRecord.RESULTS_WITH_DEFECTS)
+
+        /**
+         * Return the package from the current [OrtResult] with the given [id] or the empty package if the ID cannot be
+         * resolved. This function is useful for templates rendering advisor results, which contain only the
+         * identifiers of affected packages, but not the packages themselves.
+         */
+        fun getPackage(id: Identifier): Package =
+            input.ortResult.getPackage(id)?.pkg
+                ?: Package.EMPTY.also { log.warn { "Could not resolve package ${id.toCoordinates()}." } }
```

Comment:
Yes, I think so. When accessing a reference that evaluates to *null*, FreeMarker throws an error and aborts the generation of this report. To avoid this, the template needs conditional logic to check the value first.

Reply:
Ok, thanks. Just one nit in the log, please single-quote `'${id.toCoordinates()}'`.

Reply:
Done.